### DISCUSSION TITLE
Initial conditions are slightly modified before zeroth output

### DISCUSSION THREAD

**Discussion by msbc** [PrincetonUniversity/athena#410] [Q&A]
I'm getting that the zeroth output (outputting double-point hdf5 files) has small deviations from the initial conditions (IC) (particularly one of the velocity/momentum components) that I set in the pgen. The deviations are at the 1e-3 level, but I need them to be < 1e-5. Interestingly, these deviations are xorder dependent; the deviations from PLM to PPM are different but the same order, but keeping PLM and changing nghots doesn't seem to affect this. Other then flooring functions (which doesn't seem to be the culprit as nothing is at a floor), what could modify the IC before the zeroth output?

**Comment by pdmullen** [comment]
This is concerning.  Are you using `correct_ic=true`?  And is this for a uniform mesh?

**Reply by msbc** [reply]
So I didn't know about, `correct_ic=true`, which doesn't change the order of the deviation but enabling it makes things look like a step/sawtooth function. I am in spherical-polar coords with a logarithmic radial spacing. This issues is apparent at large R.

**Reply by felker** [reply]
This option should only be used in a very specific circumstance:
- Uniform mesh
- Cuboid cells 
- Initial condition is computed using the midpoint approximation of a smooth function

Otherwise the difference stencil will introduce unphysical oscillations, as you observed.

**Comment by c-white** [comment]
Between initialization and the first output, there are calls to functions related to turbulent driving, self gravity, 4th-order corrections, orbital advection, and diffusivity, as well as lots of internal boundary communication and prolongation/restriction. Only some of these would change the active zone conserved variables, and I don't understand how any of them depend on PLM vs PPM.

**Reply by felker** [reply]
I suggested that @msbc put a breakpoint in the reconstruction functions to see how they could possibly be called before the main timestepping loop and affect the IC. The phenomenon must be related to the use of a nonuniform, spherical mesh, but I dont see how the reconstruction affects it

**Reply by tomidakn** [reply]
In this case I would suggest to use a watch point in gdb so that you can detect when the data are modified.
